src/instructlab/model/evaluate.py

```diff
@@ -0,0 +1,504 @@
+
+# TODO: Remove after proper vLLM integration
+# pylint: disable=consider-using-with
+
+
+def get_evaluator(
+    model,
+    base_model,
+    benchmark,
+    judge_model,
+    output_dir,
+    max_workers,
+    taxonomy_path,
+    branch,
+    base_branch,
+    few_shots,
+    batch_size,
+    sdg_path,
+) -> Evaluator:
+    """takes in arguments from the CLI and uses 'benchmark' to validate other arguments
+    if all needed configuration is present, returns the appropriate Evaluator class for the benchmark
+    otherwise raises an exception for the missing values
+    """
+
+    # ensure skills benchmarks have proper arguments if selected
+    if benchmark in ["mt_bench", "mt_bench_branch"]:
+        required_args = [
+            model,
+            judge_model,
+            output_dir,
+            max_workers,
+        ]
+        required_arg_names = [
+            "model",
+            "judge-model",
+        ]
+        if benchmark == "mt_bench_branch":
+            required_args.append(taxonomy_path)
+            required_args.append(branch)
+            required_args.append(base_branch)
+            required_args.append(base_model)
+            required_arg_names.append("taxonomy-path")
+            required_arg_names.append("branch")
+            required_arg_names.append("base-branch")
+            required_arg_names.append("base-model")
+        if None in required_args:
+            click.secho(
+                f"Benchmark {benchmark} requires the following args to be set: {required_arg_names}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+        evaluator_class = BENCHMARK_TO_CLASS_MAP[benchmark]
+        if benchmark == "mt_bench":
+            return evaluator_class("test_model", "judge_model", output_dir, max_workers)
+        return evaluator_class(
+            "test_model",
+            "judge_model",
+            taxonomy_path,
+            branch,
+            output_dir,
+            max_workers,
+        )
+
+    # ensure knowledge benchmarks have proper arguments if selected
+    if benchmark in ["mmlu", "mmlu_branch"]:
+        required_args = [model, few_shots, batch_size]
+        required_arg_names = ["model"]
+        if benchmark == "mmlu_branch":
+            required_args.append(sdg_path)
+            required_args.append(base_model)
+            required_arg_names.append("sdg-path")
+            required_arg_names.append("base-model")
+        if None in required_args:
+            click.secho(
+                f"Benchmark {benchmark} requires the following args to be set: {required_arg_names}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+        evaluator_class = BENCHMARK_TO_CLASS_MAP[benchmark]
+        if benchmark == "mmlu":
+            min_tasks = os.environ.get("INSTRUCTLAB_EVAL_MMLU_MIN_TASKS")
+            if min_tasks is not None:
+                tasks = ["mmlu_abstract_algebra", "mmlu_anatomy", "mmlu_astronomy"]
```

Comment:
Is this needed as some reinforcement? If so, should you hard code the full list of 57 tasks here?

Reply:
INSTRUCTLAB_EVAL_MMLU_MIN_TASKS is in place for testing purposes to allow for a smaller list of tasks to validate the flow without running a full test.

Reply:
full list is also now a default of the MMLU evaluator class as well, so we can initialize it w/o that arg to run full-scale MMLU: https://github.com/instructlab/eval/pull/24